src/StreamMetrics.js

```diff
@@ -0,0 +1,282 @@
+
+                        for (let i = 1; i < messages.length; i++) {
+                            this.report.broker.messagesToNetworkPerSec += messages[i].broker.messagesToNetworkPerSec
+                            this.report.broker.bytesToNetworkPerSec += messages[i].broker.bytesToNetworkPerSec
+                            this.report.network.avgLatencyMs += messages[i].network.avgLatencyMs
+
+                            this.report.broker.messagesToNetworkPerSec += messages[i].broker.messagesToNetworkPerSec
+                            this.report.broker.bytesToNetworkPerSec += messages[i].broker.bytesToNetworkPerSec
+
+                            this.report.network.avgLatencyMs += messages[i].network.avgLatencyMs
+                            this.report.network.bytesToPeersPerSec += messages[i].network.bytesToPeersPerSec
+                            this.report.network.bytesFromPeersPerSec += messages[i].network.bytesFromPeersPerSec
+                            this.report.network.connections += messages[i].network.connections
+
+                            if (metricsReport.metrics['broker/cassandra']) {
+                                this.report.storage.bytesWrittenPerSec += messages[i].storage.bytesWrittenPerSec
+                                this.report.storage.bytesReadPerSec += messages[i].storage.bytesReadPerSec
+                            }
+                        }
+
+                        if (messages.length > 0) {
+                            this.report.broker.messagesToNetworkPerSec /= messages.length
+                            this.report.broker.bytesToNetworkPerSec /= messages.length
+                            this.report.network.avgLatencyMs /= messages.length
+
+                            this.report.broker.messagesToNetworkPerSec /= messages.length
+                            this.report.broker.bytesToNetworkPerSec /= messages.length
+
+                            this.report.network.avgLatencyMs /= messages.length
+                            this.report.network.bytesToPeersPerSec /= messages.length
+                            this.report.network.bytesFromPeersPerSec /= messages.length
+                            this.report.network.connections /= messages.length
+
+                            if (metricsReport.metrics['broker/cassandra']) {
+                                this.report.storage.bytesWrittenPerSec /= messages.length
+                                this.report.storage.bytesReadPerSec /= messages.length
+                            }
+
+                            if ((lastMessages[0].timestamp + this.reportMiliseconds - now) < 0) {
+                                await this.publishReport()
+                            }
+                        }
+                    }
+                }
+            }
+        } catch (e) {
+            logger.error(e)
+        }
+
+        const self = this
+        this.metricsReportTimeout = setTimeout(() => {
+            self.runReport()
```

Comment:
Shouldn't really need to use these tricks `self = this` assignments anymore with `() => {}` syntax.